Assets/Documentation/man/man3/AnimatorController.3
.TH "AnimatorController" 3 "Version 0.1" "NPP VR Simulation" \" -*- nroff -*-
.ad l
.nh
.SH NAME
AnimatorController \- This class is used to update the animator parameters based on the state of the simulation stored in an \fBNPPReactorState\fP object\&.  

.SH SYNOPSIS
.br
.PP
.PP
Inherits MonoBehaviour\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "void \fBUpdateAnimatorParameters\fP (\fBNPPReactorState\fP state)"
.br
.RI "This method updates the statemachine parameters based on the state of the simulation stored in an \fBNPPReactorState\fP object\&. "
.ti -1c
.RI "void \fBupdateScenario\fP (int scenario)"
.br
.RI "This method updates the scenario parameter of the statemachine\&. "
.ti -1c
.RI "void \fBReset\fP ()"
.br
.RI "This method resets the statemachine to the initial state\&. "
.ti -1c
.RI "int \fBgetScenario\fP ()"
.br
.RI "This method returns the scenario the state machine is currently running\&. "
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "void \fBAwake\fP ()"
.br
.RI "This method initialises animator with a reference to the statemachine \&. "
.ti -1c
.RI "\fBComponentState\fP \fBGetComponent\fP (string name, \fBComponentHealth\fP health)"
.br
.RI "This method returns the information stored in ComponentHealt for a component or initialises a new \fBComponentState\fP\&. "
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "Animator \fBanimator\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIanimator\fP is a reference to the statemachine
.RE
.PP
"
.ti -1c
.RI "\fBNPPClient\fP \fBclientScript\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
This class is used to update the animator parameters based on the state of the simulation stored in an \fBNPPReactorState\fP object\&. 
.SH "Member Function Documentation"
.PP 
.SS "void AnimatorController\&.Awake ()\fR [private]\fP"

.PP
This method initialises animator with a reference to the statemachine \&. 
.SS "\fBComponentState\fP AnimatorController\&.GetComponent (string name, \fBComponentHealth\fP health)\fR [private]\fP"

.PP
This method returns the information stored in ComponentHealt for a component or initialises a new \fBComponentState\fP\&. 
.PP
\fBParameters\fP
.RS 4
\fIname\fP is the name of a component
.br
\fIhealth\fP is a \fBComponentHealth\fP object containing the state of various components
.RE
.PP

.SS "int AnimatorController\&.getScenario ()"

.PP
This method returns the scenario the state machine is currently running\&. 
.SS "void AnimatorController\&.Reset ()"

.PP
This method resets the statemachine to the initial state\&. 
.SS "void AnimatorController\&.UpdateAnimatorParameters (\fBNPPReactorState\fP state)"

.PP
This method updates the statemachine parameters based on the state of the simulation stored in an \fBNPPReactorState\fP object\&. 
.SS "void AnimatorController\&.updateScenario (int scenario)"

.PP
This method updates the scenario parameter of the statemachine\&. 
.PP
\fBParameters\fP
.RS 4
\fIscenario\fP specifies the scenario the player has chosen
.RE
.PP

.SH "Member Data Documentation"
.PP 
.SS "Animator AnimatorController\&.animator\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIanimator\fP is a reference to the statemachine
.RE
.PP
param name="clientScript"> is a reference to the \fBNPPClient\fP instance in the scene
.SS "\fBNPPClient\fP AnimatorController\&.clientScript\fR [private]\fP"


.SH "Author"
.PP 
Generated automatically by Doxygen for NPP VR Simulation from the source code\&.


Assets/Skripte/StateMachine/AnimatorController.cs
using System.Linq;
using UnityEngine;

/// <summary>
/// This class is used to update the animator parameters based on the state of the simulation stored in an NPPReactorState object.
/// </summary>
public class AnimatorController : MonoBehaviour
{
    ///<param name="animator"> is a reference to the statemachine</param>
    private Animator animator;
    ///<param name="clientScript"> is a reference to the NPPClient instance in the scene</param>
    private NPPClient clientScript;

    /// <summary>
    /// This method initialises animator with a reference to the statemachine .
    /// </summary>
    void Awake()
    {
        clientScript = GameObject.Find("NPPclientObject").GetComponent<NPPClient>();

        animator = GetComponent<Animator>();
        if (animator == null)
        {
            Debug.LogError("Animator not found!");
        }
    }

    /// <summary>
    /// This method updates the statemachine parameters based on the state of the simulation stored in an NPPReactorState object.
    /// </summary>
    public void UpdateAnimatorParameters(NPPReactorState state)
    {
        //Set System parameter
        animator.SetBool("SimRunning", state.SystemStatus.running);
        //Set health parameter
        animator.SetBool("ReactorStatus",     GetComponent("RKS", state.ComponentHealth).broken);
        animator.SetBool("ReactorTankStatus", GetComponent("RKT", state.ComponentHealth).broken);
        animator.SetBool("CondenserStatus",   GetComponent("KNT", state.ComponentHealth).broken);
        animator.SetBool("TurbineStatus",     GetComponent("TBN", state.ComponentHealth).broken);
        animator.SetBool("WP1Status",         GetComponent("WP1", state.ComponentHealth).broken);
        animator.SetBool("WP2Status",         GetComponent("WP2", state.ComponentHealth).broken);
        animator.SetBool("CPStatus",          GetComponent("CP", state.ComponentHealth).broken);
        animator.SetBool("AtomicStatus",      GetComponent("AU", state.ComponentHealth).broken); 
        //Set valve parameter
        animator.SetBool("SV1Status", state.SV1.status);
        animator.SetBool("SV2Status", state.SV2.status);
        animator.SetBool("WV1Status", state.WV1.status);
        animator.SetBool("WV2Status", state.WV2.status);
        //Set pump parameter
        animator.SetInteger("WP1RPM", (int) state.WP1.rpm);
        animator.SetInteger("WP2RPM", (int) state.WP2.rpm);
        animator.SetInteger("CPRPM",  (int) state.CP.rpm);
        //Set reactor parameter
        animator.SetInteger("ControlRodsLvl",  (int) state.Reactor.rodPosition);
        animator.SetInteger("ReactorWaterLvl", (int) state.Reactor.waterLevel);
        animator.SetInteger("ReactorPressure", (int) state.Reactor.pressure);
        animator.SetInteger("Restheat",        (int) state.Reactor.restheat);
        //Set condenser parameter
        animator.SetInteger("CondenserWaterLvl", (int) state.Condenser.waterLevel);
        animator.SetInteger("CondenserPressure", (int) state.Condenser.pressure);
        //set generator parameter
        animator.SetInteger("GeneratorOutput", (int) state.Generator.power);

        
    }

    /// <summary>
    /// This method updates the scenario parameter of the statemachine.
    /// </summary>
    /// <param name="scenario"> specifies the scenario the player has chosen</param>
    public void updateScenario(int scenario){

        /* update the chosen scenario */

        animator.SetInteger("scenario", scenario);
    }

    /// <summary>
    /// This method resets the statemachine to the initial state.
    /// </summary>
    public void Reset(){
        animator.Play("initial", 0, 0);
    }

    /// <summary>
    /// This method returns the information stored in ComponentHealt for a component or initialises a new ComponentState.
    /// </summary>
    /// <param name="name"> is the name of a component</param>
    /// <param name="health"> is a ComponentHealth object containing the state of various components</param>

    private ComponentState GetComponent(string name, ComponentHealth health)
    {
        ComponentState component = health.components.FirstOrDefault(c => c.name == name);
        return (component == null) ? new ComponentState { name = name, status = false } : component;
    }

    /// <summary>
    /// This method returns the scenario the state machine is currently running.
    /// </summary>
    public int getScenario()
    {
        switch (animator.GetInteger("scenario"))
        {
            case 0:
                return 0;
            case 1:
                return 1;
            case 2:
                return 2;
            case 3:
                return 3;
        }

        return 0;
    }
}